src/ximea_ros_cam.cpp

```diff
@@ -206,16 +206,34 @@ void XimeaROSCam::initStorage() {
         boost::filesystem::path calib_dir = main_dir /
                                             (this->cam_name_ + "/calib/");
         this->png_path_ = calib_dir.string();
-        boost::filesystem::create_directories(calib_dir);
+        if (!boost::filesystem::create_directories(calib_dir))
+        {
+            // failed to create directory, exit ROS and explain.
+            ROS_INFO_STREAM("ERROR: unable to create directory: " << this->png_path_);
+            ROS_INFO_STREAM("Please make sure that the image_directory "
+                         << "parameter is set to a folder with the proper "
+                         << "permissions in the config file.");
+            ros::shutdown();
+        }
+
+        this->save_trigger_ = false;
```

Comment:
what does save_trigger do?

Reply:
when you press a trigger on the joystick, it saves an image